frontend/src/components/Auth/PrivateRoute.js

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { AuthContext } from '../Auth';
+
+import { SIGN_IN } from '../../constants/routes.js';
+
+/**
+ * PrivateRoute component that takes the authentication status of the current
+ * user from AuthContext. If they are authenticated, they will be allowed to
+ * view the contents of the Route component. If they are not authenticated, they
+ * will be redirected to the SIGN_IN page.
+ */
+const PrivateRoute = ({ component: RouteComponent, ...rest }) => {
+  const currentUser = useContext(AuthContext);
+
+  return (
+    <Route
+      {...rest}
+      render={routeProps =>
```

Comment:
Essentially I just want PrivateRoute to act exactly as a Route component, with the added authentication check. So yeah you have the right idea that `...rest` are just all the other props passed into `PrivateRoute`. I then used a [render prop](https://reactjs.org/docs/render-props.html) to specify that the component to be rendered is either the `RouteComponent`, if the user is logged in, or a `Redirect` component if the user is not logged in. The `routeProps` are the props that are usually passed into the `RouteComponent`, such as URL params, etc.

If you think it would be beneficial to comment my code on some of these I can definitely do that!

Reply:
I think a brief comment mentioning this would be good. +1

Reply:
Sounds good, will do that before merging.